net/redisclient.go

```diff
@@ -299,6 +299,7 @@ func (r *RedisRingClient) startUpdater(ctx context.Context) {
 		}
 
 		addrs := r.options.AddrUpdater()
+		r.metrics.UpdateGauge(r.metricsPrefix+"shards", float64(len(addrs)))
```

Comment:
Maybe we can add `r.metrics.UpdateGauge(r.metricsPrefix+"shards", float64(r.ring.Len()))` to the `StartMetricsCollection` instead?

Reply:
OTOH here it would be an instant update. Lets maybe have `r.metrics.UpdateGauge(r.metricsPrefix+"shards", float64(r.ring.Len()))` right after  `r.SetAddrs(ctx, addrs)`

Reply:
But if old and addrs is both 0 from the start the gauge will not change.
I thought it might make sense to have this independent of the internals, but I think you are right to put it 4lines down and ask the ring for the length.